source/agora/consensus/EnrollmentManager.d

```diff
@@ -633,6 +633,56 @@ public class EnrollmentManager
         return Hash.init;
     }
 
+    /***************************************************************************
+
+        Get the UTXO corresponding to the UTXO index for the height
+
+        Params:
+            height = the height of proposed block 
+            index = the index to find the UTXO with
+
+        Returns:
+            the `UTXO` corresponding to the index
+
+    ***************************************************************************/
+
+    public Hash getEnrolledUTXOByIndex (in Height height, in uint index)
+        nothrow @safe
+    {
+        Hash[] keys;
+        this.getEnrolledUTXOs(height, keys);
+        if (keys.length > index)
+            return keys[index];
+        else
+            return Hash.init;
+    }
```

Comment:
Using `Ledger.getValidators(Height)` will return a list of validators with the correct indexes.